public/app/plugins/datasource/loki/datasource.test.ts

```diff
@@ -1139,6 +1139,15 @@ describe('LokiDatasource', () => {
       });
     });
   });
+
+  describe('getQueryStats', () => {
+    it('uses statsMetadataRequest', async () => {
+      const ds = createLokiDatasource(templateSrvStub);
+      const spy = jest.spyOn(ds, 'statsMetadataRequest').mockResolvedValue({} as QueryStats);
+      ds.getQueryStats('{foo="bar"}');
+      expect(spy).toHaveBeenCalled();
+    });
+  });
```

Comment:
Sorry to be annoying. If you could add another case for expecting to throw it'd be ideal:

```
    if (url.startsWith('/')) {
      throw new Error(`invalid metadata request url: ${url}`);
    }
```

Reply:
Good idea! Added. :)

Reply:
Fantastic! Thanks 😄